### ISSUE TITLE
DeepSeek Preffil Pcc analysis

### ISSUE THREAD

**Issue by jmitrovicTT** [tenstorrent/tt-metal#44475]
### tt-metal dataset
- 164443 DS tokens
- .py, .yaml, .cpp

| Title | Image |
|---|---|
| **Device Gate** | <img width="800" height="312" alt="Host Gate" src="https://github.com/user-attachments/assets/113c9b4d-6f2a-47c0-add6-34ebb9c8487d" /> |
| **Host Gate** | <img width="800" height="312" alt="Device Gate" src="https://github.com/user-attachments/assets/9c63ec23-4eaa-4508-8ba5-51cf6c8abe34" /> |

- Device: `Failed: PCC below 0.97 at: layer_59: 0.963629; layer_60: 0.887291`
    - logits: `0.997943` PASS✅
- Host: `passes`
    - logits: `0.997389` PASS✅

### Hardy, A Course of Pure Mathematics
- 131479 DS tokens
- LaTeX

| Title | Image |
|---|---|
| **Device Gate** | <img width="800" height="312" alt="Device Experts" src="https://github.com/user-attachments/assets/f72e89b7-df3a-4f1a-95dc-4e477b6476de" /> |
| **Host Gate** | <img width="800" height="312" alt="Host Experts" src="https://github.com/user-attachments/assets/23a2ad42-1502-430c-8c62-58dd9cdb4b40" /> |

- Host: `Failed: PCC below 0.97 at: layer_54: 0.969832; layer_55: 0.969979; layer_56: 0.969632; layer_57: 0.969974; layer_60: 0.914326`
    - logits: `0.978495`      PASS✅
- Device: `Failed: PCC below 0.97 layer_45: 0.969461; layer_46: 0.968678; layer_47: 0.967288; layer_48: 0.966751; layer_49: 0.965624; layer_50: 0.964052; layer_51: 0.963656; layer_52: 0.963630; layer_53: 0.964511; layer_54: 0.963480; layer_55: 0.963761; layer_56: 0.963375; layer_57: 0.963846; layer_58: 0.964576; layer_59: 0.966192; layer_60: 0.921799`
    - logits: `0.973434`      PASS✅

## HeatMaps of experts (GPU)

| tt-metal | Math |
|---|---|
| <img width="800" height="800" alt="Image" src="https://github.com/user-attachments/assets/651f6e70-2907-44cd-99d2-33545ba22ec3" /> | <img width="800" height="800" alt="Image" src="https://github.com/user-attachments/assets/3d241d76-dc9b-4a60-864b-016f0561778f" /> |

| Longbook | Math |
|---|---|
| <img width="800" height="800" alt="Image" src="https://github.com/user-attachments/assets/04fc94b0-f0e6-4df6-83dd-b638055d306e" /> | <img width="800" height="800" alt="Image" src="https://github.com/user-attachments/assets/3d241d76-dc9b-4a60-864b-016f0561778f" /> |

## 1) DeepSeek Prefill 25600 tokens

**- e256_host, pure_math, 25600:**

```
layer_58                0.970674      PASS
layer_59                0.972114      PASS
layer_60                0.938307      FAIL
logits                  0.997749      PASS
```

**- e256_device_fp32, 61_layers-25600, pure_math_25600**

```
layer_58                0.969714      PASS
layer_59                0.971261      PASS
layer_60                0.918836      FAIL
logits                  0.998128      PASS
```

**- e256_device, 61_layers , pure_math, 25600**

```
layer_58                0.964350      PASS
layer_59                0.966127      PASS
layer_60                0.908283      PASS
logits                  0.995721      PASS
```

- **e256_host, 61_layers-tt_metal_text, 25600**

```
layer_58                0.979829      PASS
layer_59                0.970167      PASS
layer_60                0.924455      FAIL
logits                  0.997992      PASS
```

- **e256_device_fp32, 61_layers, tt_metal_text_25600**

```
layer_58                0.979408      PASS
layer_59                0.970200      PASS
layer_60                0.933474      FAIL
logits                  0.998152      PASS
```
**- e256_device, 61_layers-25600, tt_metal_text**

```
layer_58                0.977154      PASS
layer_59                0.970154      PASS
layer_60                0.926304      PASS
logits                  0.997832      PASS
```

| tt-metal | Math |
|---|---|
| <img width="800" height="800" alt="Image" src="https://github.com/user-attachments/assets/e9d9a2c1-31de-4ca0-a28d-3c4ce0621e3b" /> | <img width="800" height="800" alt="Image" src="https://github.com/user-attachments/assets/689702fc-08b1-43a8-bf8c-489aaf56ed5a" /> |

# Experiment: Override Gate experts Ids with GPU ids

### 1) **PCC results**

**- pure_math_25600**

```
layer_58                0.983616      PASS
layer_59                0.984375      PASS
layer_60                0.989977      PASS
logits                  0.996920      PASS
```

**- tt_metal_text_25600**

```
layer_58                0.990827      PASS
layer_59                0.972092      PASS
layer_60                0.994131      PASS
logits                  0.999338      PASS
```

# Experiment: **Experts overlap analysis**

|                | Math (`pure_math_25600`) | Metal (`tt_metal_text`) |
|----------------|--------------------------|--------------------------|
| **Host** | overall mean: **0.850841**<br>MIN: layer 57 = 0.776387<br>MAX: layer 3 = 0.964233 | overall mean: **0.846792**<br>MIN: layer 48 = 0.794209<br>MAX: layer 3 = 0.965786 |
| **Device** | overall mean: **0.766693**<br>MIN: layer 48 = 0.629790<br>MAX: layer 21 = 0.854009 | overall mean: **0.781190**<br>MIN: layer 48 = 0.653179<br>MAX: layer 60 = 0.861299 |

<img width="800" height="390" alt="Image" src="https://github.com/user-attachments/assets/25ba1fae-8aec-4d10-b167-dd2d110c0b5b" />
